services/discourse_translator/google.rb

```diff
@@ -75,9 +75,12 @@ def self.access_token
     end
 
     def self.detect(topic_or_post)
+      text =
+        topic_or_post.class.name == "Post" ? topic_or_post.cooked : topic_or_post.first_post&.cooked
+
       topic_or_post.custom_fields[DiscourseTranslator::DETECTED_LANG_CUSTOM_FIELD] ||= result(
         DETECT_URI,
-        q: get_text(topic_or_post).truncate(MAXLENGTH, omission: nil),
```

Comment:
The `get_text` helper is used in all the translator services (Google, Amazon, etc.), so it would be good for the fix here to be moved up a level, into `get_text` ideally. 

That helper was recently updated by @lis2 in https://github.com/discourse/discourse-translator/pull/130/files. 

I think what is happening is that the title translation is attempted after trying out the post but it is erroring (because of `en|en`) and the `translate` endpoint returns a 422. 

It may be best to just ignore the error when users have topics with title in same language as the requested translation (and body in a different language). That _should_ allow the post translation to then be returned properly. 

Switching to the first post detection just for Google has two issues: 1) it might give you the wrong language for the topic title and 2) it applies only to the Google translation service, leaving the others exposed to the same problem.

Reply:
thanks penar. in digging back in to this, I realized my approach here is wrong altogether ... I made a new PR with a different approach. won't burden you with it Friday at 5pm. Have a great weekend!